src/nupic/ntypes/Sdr.hpp

```diff
@@ -154,11 +165,13 @@ class SparseDistributedRepresentation : public Serializable
     };
 
     /**
-     * TODO: This comment
+     * Notify everyone that this SDR's value has officially changed.
      */
     void do_callbacks() {
-        for(const auto func_ptr : callbacks)
-            func_ptr();
+        for(const auto func_ptr : callbacks) {
+            if( func_ptr != NULL )
+                func_ptr();
```

Comment:
do we want a try-catch, so an err in ahook does not crash the parent SDR?

Reply:
But if there is an error in a callback hook and the SDR catches it, then where should the error go?  Surely it shouldn't be silenced but rather raised to the attention of the developer who made the hook.

 Also I think this sort of error is unrecoverable from, since callbacks can be critical to program function.